example/21_gemm_layernorm/gemm_xdl_layernorm_single_kernel_fp16.cpp

```diff
@@ -0,0 +1,252 @@
+#include <iostream>
+#include <numeric>
+#include <initializer_list>
+#include <cstdlib>
+#include <stdlib.h>
+#include <half.hpp>
+#include "check_err.hpp"
+#include "config.hpp"
+#include "device.hpp"
+#include "host_tensor.hpp"
+#include "host_tensor_generator.hpp"
+#include "device_tensor.hpp"
+#include "tensor_layout.hpp"
+#include "device_gemm_xdl_layernorm_cshuffle.hpp"
+#include "element_wise_operation.hpp"
+#include "reduction_operator.hpp"
+#include "reference_gemm_layernorm.hpp"
+#include "gemm_specialization.hpp"
+
+template <ck::index_t... Is>
+using S = ck::Sequence<Is...>;
+
+using F16 = ck::half_t;
+using F32 = float;
+
+using Row = ck::tensor_layout::gemm::RowMajor;
+using Col = ck::tensor_layout::gemm::ColumnMajor;
+
+using ADataType   = F16;
+using BDataType   = F16;
+using CDataType   = F16;
```

Comment:
How about adding bias, gamma and beta type?
Especially in int8 implementation, bias would be same as acc type.

Reply:
I have added extra precision type for C0 which applies to bias/gamma/beta.

Reply:
Because in this class, you already have CLayout, CDataType, c_grid_desc,  CShuffleXXXXX, CReduceThreadXXXX, block_2_ctile_map...and so on.
Hence, I thought naming of CElementwiseOperation and p_c_grid_ (global mem) might be confusing.